opentelemetry-otlp/src/lib.rs

```diff
@@ -119,27 +119,22 @@
 
 use opentelemetry::{global, sdk, trace::TracerProvider};
 
-#[cfg(all(feature = "grpc-sys", not(feature = "tonic")))]
+#[cfg(feature = "grpc-sys")]
 use std::collections::HashMap;
 
 use std::str::FromStr;
 use std::time::Duration;
 
-#[cfg(all(feature = "tonic", not(feature = "integration-testing")))]
+#[cfg(
+not(feature = "integration-testing")
+)]
 #[rustfmt::skip]
-#[allow(clippy::all, unreachable_pub)]
-mod proto;
-
-#[cfg(all(
-    feature = "grpc-sys",
-    not(feature = "tonic"),
-    not(feature = "integration-testing")
-))]
-#[allow(clippy::all, unreachable_pub, dead_code)]
+#[allow(warnings)]
```

Comment:
Why are we allowing all the warnings here? Maybe add a comment about this?

Reply:
Mostly because the proto mod only contains generated code. We should change it.